docs/recipes/Code-Splitting-Using-Holocron.md

```diff
@@ -1,7 +1,238 @@
+import './Chunk';
+
+import(/* webpackChunkName: "<chunkName>" */ './Chunk');
+```
+
+Our _chunk_ can be anything from a component to a `node_modules` package. Once we run `bundle-module` from `one-app-bundler`, our output in `build/<version>` will contain
+additional JavaScript files based on the number of chunks we have dynamically split
+from our module.
+
+```
+root
+└── build
+    └── 1.0.0
+        ├── <chunkName>.<moduleName>.chunk.browser.js
+        ├── vendors~<chunkName>.<moduleName>.chunk.browser.js
+        ├── <chunkName>.<moduleName>.chunk.legacy.browser.js
+        ├── <moduleName>.browser.js
+        ├── <moduleName>.legacy.browser.js
+        ├── <moduleName>.node.js
+        └── en-us
+            ├── <moduleName>.json
+            ├── integration.json
+            └── qa.json
+```
+
+### Using **`holocronModule`**
+
+`src/Module.jsx`
+```jsx
+import React from 'react';
+import { holocronModule } from 'holocron';
+
+let Chunk = () => <p>Loading...</p>;
+
+export function MyHolocronModule() {
+  return <Chunk />;
+}
+
+export default holocronModule({
+  name: 'holocron-module',
+  load: () => () => Promise.all([
+    import(/* webpackChunkName: '<chunkName>' */ './Chunk')
+      .then((imported) => imported.default || imported)
+      .then((Component) => {
+        Chunk = Component;
+      })
+      .catch((error) => {
+        Chunk = () => <p>{error.message}</p>;
+      }),
+    //  ... load language packs, etc
+  ]),
+  options: { ssr: true },
+})(MyHolocronModule);
+```
+### Using `Suspense` and `lazy`
+
+> server side rendering is not compatible with Suspense & lazy
+
+`src/Module.jsx`
+```jsx
+import React from 'react';
+
+const Chunk = React.lazy(() => import(/* webpackChunkName: "<chunkName>" */ './Chunk')
+  .then((importedChunk) => importedChunk.default || importedChunk));
+
+export default function MyHolocronModule() {
+  return (
+    <React.Suspense fallback={<p>Loading...</p>}>
+      <Chunk />
+    </React.Suspense>
+  );
+}
+```
+
+### How it works
+
+One App uses [`one-app-bundler`][one-app-bundler-readme] to support dynamic importing and code
+plitting on the module level. Out of the box, [`one-app-bundler`][one-app-bundler-readme] uses
+`webpack` and `babel` (with `@babel/plugin-syntax-dynamic-import`) under the hood to understand
+the `import()` syntax and automatically split your module where ever the `import() ` statement
+is used within your module.
+
+We can expect the final bundled module output to contain everything it nornally does,
+with two additional JavaScript files per chunk (modern `browser` and `legacy.rowser` versions).
+For the `node` build output, all the dynamic imports have been imported, bundled and treated
+as a `webpack_require` call.
+
+### Include your existing tools to the pipeline
+
+It can be helpful to test your current apps for migration, supporting react-loadable and @loadable/component. Extend the webpack config if you need.
```

Comment:
```suggestion
It can be helpful to test your current apps for migration, supporting react-loadable and
@loadable/component. Extend the webpack config if you need.
```